src/toxcore/onion/packet/mod.rs

```diff
@@ -258,7 +258,7 @@ impl ToBytes for OnionReturn {
 }
 
 impl OnionReturn {
-    #[cfg_attr(feature = "cargo-clippy", allow(clippy::needless_pass_by_value))]
+    #[allow(clippy::needless_pass_by_value)]
```

Comment:
Why needless_pass_by_value? Can you change inner to `&Option<OnionReturn>` using Option::as_ref in the call site?

Reply:
It's not the Option that gives this warning, it's `buf: (&'a mut [u8], usize)`. Clippy suggest to pass it by ref as `buf: &(&'a mut [u8], usize)` which is not very idiomatic for `cookie-factory` (however doable). Shall I change it?

Reply:
Ah I see. No, keep it.